Client/Assets/SessionRestore.html

```diff
@@ -17,10 +17,6 @@
    */
   (function () {
       function getRestoreURL(url) {
-          // If the URL is already an internally hosted page, we can restore it directly.
-          if (url.indexOf(document.location.origin) === 0) {
-              return url;
-          }
           // Otherwise, push an error page to trigger a redirect when loaded.
```

Comment:
> The change in SessionRestore.html was necessary because if you navigated back to the home page and then restarted the browser (so you have some history). Your history would be available but when you went forward the page would not load. I have no freaking clue why. Webkit would just refuse to load error/error.html with no errors or callbacks

Yeah, strange...I can reproduce this too. One of the reasons we have this check is to prevent error pages from snowballing each time the browser is restored; i.e., close/reopen the browser several times, and a new layer of `http://localhost:6571/errors/error.html...` will be prepended to each page each time.

It seems like we need to load `error.html` for session restore to work, so how about this: rather than using the URL as-is for all localhost origins, only use the URL as-is if it's an error page (that is, check `document.location.origin` *and* the path). That way, we're always guaranteed to load an error page, and they won't pile on themselves.